### PR TITLE
BERT calibration and QDQ quantization sample for TensorRT

### PR DESCRIPTION
**Pull request by stevenlix**
BERT QDQ Quantization for TensorRT.
    Step 1, calibrate original FP32 BERT model based on SQuAD dataset to get dynamic range
    Step 2, insert Q/DQ nodes with scale and zero-point to get QDQ model

    The original FP32 BERT model is converted from Hugging Face BERT model,
    https://huggingface.co/transformers/serialization.html#converting-an-onnx-model-using-the-transformers-onnx-package
    Some utility functions for dataset processing and data reader are from Nvidia TensorRT demo BERT repo,
    https://github.com/NVIDIA/TensorRT/tree/master/demo/BERT

### CODE REVIEW

File: onnxruntime/python/tools/quantization/E2E_example_model/nlp/bert/trt/e2e_tensorrt_bert_example.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    model_path = "./model.onnx"
+    squad_json = "./dev-v1.1.json"
+    vocab_file = "./vocab.txt"
+    calibration_cache_file = "./bert.flatbuffers"
+    augmented_model_path = "./augmented_model.onnx"
+    qdq_model_path = "./qdq_model.onnx"
+    sequence_lengths = [384]
+    calib_num = 100
+
+    # Generate INT8 calibration cache
```
Comment (yufenglee) [inline onnxruntime/python/tools/quantization/E2E_example_model/nlp/bert/trt/e2e_tensorrt_bert_example.py:97]: Just call quantizat_static?

**Comment by yufenglee** [comment]
We may move all the examples to https://github.com/microsoft/onnxruntime-inference-examples.

File: onnxruntime/python/tools/quantization/E2E_example_model/nlp/bert/trt/e2e_tensorrt_bert_example.py (hunk 1 above)
Comment (stevenlix) [reply onnxruntime/python/tools/quantization/E2E_example_model/nlp/bert/trt/e2e_tensorrt_bert_example.py:97]: We also want to save compute range to calibration table for later use in TRT, which is missing in quantize_static.

**Comment by stevenlix** [comment]
> We may move all the examples to https://github.com/microsoft/onnxruntime-inference-examples.

Agree. Is there a timeline to do this? If it's after ORT1.9 release. Let's merge the PR to master first, then move all the e2e examples together later.

**Comment by jywu-msft** [comment]
> > We may move all the examples to https://github.com/microsoft/onnxruntime-inference-examples.
> 
> Agree. Is there a timeline to do this? If it's after ORT1.9 release, let's merge the PR to master first, then move all the e2e examples together later.

I think it's better to merge this new sample into the final location now than have to move it soon after.
we can start a separate PR to move all quantization samples to the new samples repo @yufenglee , can you do it soon?
or I can help do it as well.

**Comment by stevenlix** [comment]
Close the PR. A new PR is created in the onnxruntime-inference-examples repo, https://github.com/microsoft/onnxruntime-inference-examples/pull/28

**Pull request closed without merging**